lib/fastMerge.js

```diff
@@ -1,99 +1,45 @@
-* @param optionsArgument
-* @returns {*}
-*/
-let valArr;
-function defaultArrayMerge(target, source, optionsArgument) { 
-  var destination = target.slice();
-  for (let i = 0; i < source.length; ++i) {
-    valArr = source[i];
-    if (i >= destination.length) {
-      destination.push(cloneIfNecessary(valArr, optionsArgument));
-    } else if (typeof destination[i] === 'undefined') {
-      destination[i] = cloneIfNecessary(valArr, optionsArgument);
-    } else if (isMergeableObject(valArr)) {
-      destination[i] = merge(target[i], valArr, optionsArgument);
+
+function mergeObject(target, source) {
+    const destination = {};
+    if (isMergeableObject(target)) {
+        // eslint-disable-next-line rulesdir/prefer-underscore-method
+        const targetKeys = Object.keys(target);
+        for (let i = 0; i < targetKeys.length; ++i) {
+            const key = targetKeys[i];
+            destination[key] = target[key];
+        }
     }
-  }
-  return destination;
-}
-/**
-* Merge Object
-*
-* @param target
-* @param source
-* @param optionsArgument
-* @returns {{}}
-*/
-function mergeObject(target, source, optionsArgument) {
-  var destination = {};
-  if (isMergeableObject(target)) {
-    Object.keys(target).forEach(function (key) {
-      destination[key] = cloneIfNecessary(target[key], optionsArgument)
-    })
-  }
-  Object.keys(source).forEach(function (key) {
-    if (!isMergeableObject(source[key]) || !target[key]) {
-      destination[key] = cloneIfNecessary(source[key], optionsArgument)
-    } else {
-      destination[key] = merge(target[key], source[key], optionsArgument)
+    // eslint-disable-next-line rulesdir/prefer-underscore-method
+    const sourceKeys = Object.keys(source);
+    for (let i = 0; i < sourceKeys.length; ++i) {
+        const key = sourceKeys[i];
+        if (source[key] === undefined) {
+            // noop
+        } else if (!isMergeableObject(source[key]) || !target[key]) {
+            destination[key] = source[key];
+        } else {
+            // eslint-disable-next-line no-use-before-define
+            destination[key] = fastMerge(target[key], source[key]);
+        }
     }
-  });
-  return destination
+
+    return destination;
 }
-/**
- * Merge Object and Arrays
- *
- * @param target
- * @param source
- * @param optionsArgument
- * @returns {*}
- */
-export function merge(target, source, optionsArgument) {
-  var array = Array.isArray(source);
-  var options = optionsArgument || { arrayMerge: defaultArrayMerge };
-  var arrayMerge = options.arrayMerge || defaultArrayMerge;
-  if (array) { // We may not even need that as mergeWithCustomized suggests it
-    return Array.isArray(target) ? arrayMerge(target, source, optionsArgument) : cloneIfNecessary(source, optionsArgument);
-  } else {
-    return mergeObject(target, source, optionsArgument);
-  }
+
+// eslint-disable-next-line import/prefer-default-export, rulesdir/no-inline-named-export
+export function fastMerge(target, source) {
```

Comment:
I think using a default export here is fine (rather than disabling the lint rule) since there is only one thing being exported.
```suggestion
export default (target, source) => {
```